salt/pillar/__init__.py

```diff
@@ -761,7 +761,7 @@ def render_pstate(self, sls, saltenv, mods, defaults=None):
                                      '\'{0}\' found'.format(saltenv)]
                                 )
 
-                        for sub_sls in set(matched_pstates):
+                        for sub_sls in matched_pstates:
```

Comment:
These will still need to be uniqued, because it is possible for multiple states to include the same state.

Reply:
@isbm 
```
[_matched_pstates.append(el) for el in matched_pstates if el not in _matched_pstates]
```

Will keep the first item of duplications, but I think in isse #47954 , @steverweber expected keep the last one of duplications.

Reply:
I noticed that. Maybe I am over-thinking about this case:

```
>>> matched_pstates = ['a', 'b', 'a']
>>> _matched_pstates = []
>>> [_matched_pstates.append(el) for el in matched_pstates if el not in _matched_pstates]
>>> _matched_pstates
['a', 'b']
>>>
```
Which comes to a my consideration that this `a` state should be keep before `b` or after?

Reply:
Yes, I was thinking about overring/merge policy. Later of this block of code theres was an option called "pillar_includes_override_sls", maybe this unique policy should also provided as an option with default value set, so users can choice to use unique or override or merge policy.